### ISSUE TITLE
Automatic code formatting?

### ISSUE THREAD

**Issue by jowr** [CoolProp/CoolProp#1254]
Should we somehow integrate ClangFormat into our workflow? It could be nice to unify code formatting and to make sure that new code complies with the specified format.

**Comment by jowr** [comment]
Have a look at: http://clang.llvm.org/docs/ClangFormat.html

**Comment by ibell** [comment]
Especially if we can ensure that code doesn't use tabs instead of spaces, I
would be interested.  How strict is the formatting checking?  Do you
recommend one style in particular?

On Oct 6, 2016 2:00 AM, "Jorrit Wronski" <notifications@github.com> wrote:

Have a look at: http://clang.llvm.org/docs/ClangFormat.html

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/CoolProp/CoolProp/issues/1254#issuecomment-251893214>,
or mute the thread

**Comment by mmoelle1** [comment]
As a side remark, I use the LLVM default format and slightly adjusted it to what I like my source codes to look like. Tabs are replaced by spaces. If you store the .clang-format file in you git-repo you can even install a git-hook that basically auto-formats files before commit.

**Comment by jowr** [comment]
Yes, good point. We might want to use https://clangformat.com/ to build our format definitions.

**Comment by ibell** [comment]
I'm fine with someone taking the lead on this.  For me the only thing that I *really* care about is tabs being replaced with spaces and I can live with most everything else

**Comment by thorade** [comment]
For the tabs/spaces thing, an editorconfig file should be sufficient:
http://editorconfig.org/

**Comment by thorade** [comment]
The Python files could be formatted using the autopep8 script.
https://pypi.python.org/pypi/autopep8

**Comment by jowr** [comment]
I really like the editorconfig thingy - can you make a style-file?

**Comment by ibell** [comment]
Yeah I love this idea.  Can we get it to autofix the code?  The tabs/spaces
is the most pernicious issue in my experience

**Comment by thorade** [comment]
So far, I have only seen this, but never used it, nor written a file, but it should not be too hard. I'll create a minimal file that can be extended later, e.g. for other languages, more styles...
So, a tab should be four spaces, in  all languages?

For fixing the existing code: Python code can be fixed with autopep8, C++ code can probably be fixed by an advanced IDE, like [CLion from Jetbrains](https://www.jetbrains.com/clion/features/).
https://www.jetbrains.com/help/clion/changing-indentation.html

**Comment by scls19fr** [comment]
For Python code black could help https://black.readthedocs.io/en/stable/

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
